common/src/commonMain/kotlin/com/cuhacking/atlas/common/SearchViewModel.kt

```diff
@@ -0,0 +1,23 @@
+package com.cuhacking.atlas.common
+
+import com.cuhacking.atlas.db.Database
+import com.cuhacking.atlas.db.Feature_fts
+import kotlinx.coroutines.coroutineScope
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+
+class SearchViewModel(private val database: Database) {
+
+    private val _searchResults: MutableStateFlow<List<SearchResult>> = MutableStateFlow(listOf())
+    val searchResults: StateFlow<List<SearchResult>> = _searchResults
+
+    suspend fun getSearchResults(query: String) = coroutineScope {
```

Comment:
```suggestion
    suspend fun getSearchResults(query: String) = withContext(dispatchers.io) {
```

I didn't really touch on this last week (so this isn't something I was expecting you to have done), but what this does is it will cause this suspend function to run on a thread dedicated for doing I/O (like reading a database). 

What coroutines uses to switch thread contexts (hence the name `withContext`) are called "dispatchers". The `CoroutineDispatchers` object is defined in [CoroutineDispatchers.kt](common/src/commonMain/kotlin/com/cuhacking/atlas/common/CoroutineDispatchers.kt) and it just maps some coroutine dispatcher to a name we can recognize since each platform has different dispatchers available. 

Android has a dedicated I/O dispatcher, iOS doesn't, and JS is single threaded so there's only really one dispatcher..